src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -780,20 +781,28 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 
 	//#endregion
 
-	//#region invokeWithinEditorContext()
+	//#region getEditorContextKeyService()
 
-	invokeWithinEditorContext<T>(fn: (accessor: ServicesAccessor) => T): T {
-		const activeTextEditorControl = this.activeTextEditorControl;
-		if (isCodeEditor(activeTextEditorControl)) {
-			return activeTextEditorControl.invokeWithinContext(fn);
+	getEditorContextKeyService(): IContextKeyService {
+		const activeEditorPane = this.activeEditorPane;
+		if (activeEditorPane) {
+			const result = activeEditorPane.getInternalContextKeyService();
+			if (result) {
+				return result;
+			}
+
+			const activeControl = activeEditorPane.getControl();
```

Comment:
This now feels to me like it is at the wrong layer. Implementors of editors should do the right thing, I would expect the editor service to not have any smarts.